tests/test_gdal_subsetter.py

```diff
@@ -0,0 +1,94 @@
+""" A module for end-to-end tests of the Harmony GDAL Adapter. """
+from datetime import datetime
+from os.path import exists, join as path_join
+from pathlib import Path
+from shutil import rmtree
+from tempfile import mkdtemp
+from unittest import TestCase
+from unittest.mock import patch
+
+from harmony.message import Message
+from harmony.util import config as harmony_config
+from pystac import Asset, Catalog, Item
+
+from gdal_subsetter.exceptions import DownloadError, UnknownFileFormatError
+from gdal_subsetter.transform import HarmonyAdapter
+
+
+class TestEndToEnd(TestCase):
```

Comment:
I think, eventually, the tests in `test_transform.py` and `test_transform_no_download.py` could be migrated into more of this sort of (traditional `unittest`) structure. They probably would be merged with this test class.